Userland/Applications/SoundPlayer/main.cpp

```diff
@@ -125,7 +129,21 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     visualization_actions.add_action(samples);
 
     auto album_cover_visualization = GUI::Action::create_checkable("&Album Cover", [&](auto&) {
-        static_cast<SoundPlayerWidgetAdvancedView*>(player)->set_visualization<AlbumCoverVisualizationWidget>();
+        auto get_image_from_music_file = [&player, &decoder_client]() -> RefPtr<Gfx::Bitmap> {
+            auto const& pictures = player->pictures();
+
+            if (pictures.is_empty())
+                return {};
+
+            auto decoded_image_or_error = decoder_client->decode_image(pictures[0].data);
```

Comment:
Do we always want the first picture? or could there be one picture per track in this data? Maybe this is just a FIXME for later to come back and make the selected picture depend on the current track.

Reply:
At best, it should be an option for the user, but I don't even know a player that ships this functionality.
This data is per track, so if each track of an album contains a different cover image, it will update automatically.

I added a fixme for clarification.